docs/content/docs/dev/table/functions/ptfs.md

```diff
@@ -1093,6 +1093,359 @@ This allows the PTF to focus on the main aggregation without the need to manuall
+- Thus, every message is either an insertion (`+`) or its retraction (`-`).
+- The mode is known as *full image* in the literature.
+
+### Updating Input Tables
+
+The `ArgumentTrait.SUPPORTS_UPDATES` instructs the system that updates are allowed as input to the given table argument.
+By default, a table argument is insert-only and updates will be rejected.
+
+Input tables become updating when sub queries such as aggregations or outer joins force an incremental computation. For
+example, the following query only works if the function is able to digest retraction messages:
+
+```text
+// The change +I[1] followed by -U[1], +U[2], -U[2], +U[3] will enter the function
+WITH UpdatingTable AS (
+  SELECT COUNT(*) FROM (VALUES 1, 2, 3)
+)
+SELECT * FROM f(table_arg => TABLE UpdatingTable)
+```
+
+If updates should be supported, ensure that the data type of the table argument is chosen in a way that it can encode
+changes. In other words: choose a `Row` type that exposes the `RowKind` change flag.
+
+The changelog of the backing input table decides which kinds of changes enter the function. The function receives `{+I}`
+when the input table is append-only. The function receives `{+I,+U,-D}` if the input table is upserting using the same
+upsert key as the partition key. Otherwise, retractions `{+I,-U,+U,-D}` (i.e. including `RowKind.UPDATE_BEFORE`) enter
+the function. Use `ArgumentTrait.REQUIRE_UPDATE_BEFORE` to enforce retractions for all updating cases.
+
+For upserting tables, if the changelog contains key-only deletions (also known as partial deletions), only upsert key
+fields are set when a row enters the function. Non-key fields are set to `null`, regardless of `NOT NULL` constraints.
+Use `ArgumentTrait.REQUIRE_FULL_DELETE` to enforce that only full deletes enter the function.
+
+The `SUPPORTS_UPDATES` trait is intended for advanced use cases. Please note that inputs are always insert-only in batch
+mode. Thus, if the PTF should produce the same results in both batch and streaming mode, results should be emitted based
+on watermarks and event-time.
+
+#### Enforcing Retract Mode
+
+The `ArgumentTrait.REQUIRE_UPDATE_BEFORE` instructs the system that a table argument which `SUPPORT_UPDATES` should include
+a `RowKind.UPDATE_BEFORE` message when encoding updates. In other words: it enforces presenting the updating table in
+retract changelog mode.
+
+By default, updates are encoded as emitted by the input operation. Thus, the updating table might be encoded in upsert
+changelog mode and deletes might only contain keys.
+
+The following example shows how the input changelog encodes updates differently:
+
+```text
+// Given a table UpdatingTable(name STRING PRIMARY KEY, score INT)
+// backed by upsert changelog with changes
+// +I[Alice, 42], +I[Bob, 0], +U[Bob, 2], +U[Bob, 100], -D[Bob, NULL].
+
+SELECT * FROM f(table_arg => TABLE UpdatingTable PARTITION BY name)
```

Comment:
nit: you may mention in a comment that this ptf `f` uses `REQUIRE_UPDATE_BEFORE` and `REQUIRE_FULL_DELETE`